src/roswire/ros1/package.py

```diff
@@ -1,29 +1,110 @@
 # -*- coding: utf-8 -*-
-__all__ = ("ROS1PackageDatabase",)
+__all__ = ("ROS1Package", "ROS1PackageDatabase",)
 
 import os
 import typing
-from typing import Any, Dict, Iterable, List
+from typing import Any, Collection, Dict, List
+from typing import Iterable  # noqa: F401 # Needed for tuple_from_iterable
 
+import attr
 import dockerblade
 from loguru import logger
+from roswire.common import ActionFormat, MsgFormat, SrvFormat
+from roswire.util import tuple_from_iterable
 
 from ..common import Package, PackageDatabase
 
 if typing.TYPE_CHECKING:
     from .. import AppInstance
 
 
-class ROS1PackageDatabase(PackageDatabase):
+@attr.s(frozen=True, auto_attribs=True, slots=True)
+class ROS1Package(Package[MsgFormat, SrvFormat, ActionFormat]):
+    name: str
+    path: str
+    messages: Collection[MsgFormat] = attr.ib(converter=tuple_from_iterable)
+    services: Collection[SrvFormat] = attr.ib(converter=tuple_from_iterable)
+    actions: Collection[ActionFormat] = attr.ib(converter=tuple_from_iterable)
+
+    @classmethod
+    def build(cls, path: str, app_instance: "AppInstance") -> "ROS1Package":
```

Comment:
This violates LSP. (Return type of `build` in `Package` is `Package`.) We'll need to think about how to deal with this one.

Reply:
One way to avoid this issue is to move this logic into the `_build_package` method of `PackageDatabase`.